src/core/mesh/qgsmeshrenderersettings.cpp

```diff
@@ -340,6 +340,23 @@ void QgsMeshRendererVectorArrowSettings::readXml( const QDomElement &elem )
 
 // ---------------------------------------------------------------------
 
+QgsMeshRendererSettings::QgsMeshRendererSettings()
+  : mAveragingMethod( new QgsMeshSingleLevelAveragingMethod() )
```

Comment:
isn't it better to default to averaging method == nullptr ? and only if the mesh is 3d the mesh layer constructor would set a default averaging method?

Reply:
it isn't that easy to see if the mesh is 3d. 3d is defined on datasets, so one would need to go through all datasets to find what is going on. I prefer to keep it this way